src/webgpu/shader/execution/builtin/isnan.spec.ts

```diff
@@ -0,0 +1,77 @@
+export const description = `WGSL execution test. Section: Value-testing built-in functions. Function: isNan`;
+
+import { makeTestGroup } from '../../../../common/framework/test_group.js';
+import { GPUTest } from '../../../gpu_test.js';
+import { TypeBool, TypeF32, f32, f32Bits, False, True } from '../../../util/conversion.js';
+import { subnormalF32Examples, normalF32Examples, nanF32BitsExamples } from '../../values.js';
+
+import { kBit, run } from './builtin.js';
+
+export const g = makeTestGroup(GPUTest);
+
+g.test('value_testing_builtin_functions,isNan')
+  .uniqueId('fdd1e7105af70b74')
+  .specURL('https://www.w3.org/TR/2021/WD-WGSL-20210929/#value-testing-builtin-functions')
+  .desc(
+    `
+isNan:
+I is f32 or vecN<f32> T is bool if I is a scalar, or vecN<bool> if I is a vector isNan(e: I ) -> T Test for NaN according to IEEE-754. Component-wise when I is a vector. (OpIsNan)
+
+Please read the following guidelines before contributing:
+https://github.com/gpuweb/cts/blob/main/docs/plan_autogen.md
+`
+  )
+  // https://gpuweb.github.io/gpuweb/wgsl/#floating-point-evaluation says:
+  //
+  //    Implementations may assume that NaNs, inifinities are not present
+  //
+  //       Note: This means some functions (e.g. isInf, isNan, min and max) may not return
+  //       the expected result due to optimizations about the presence of NaNs and infinities.
+  //
+  // TODO: For now, test assuming NaNs are handled strictly, according to IEEE-754. Relax
```

Comment:
Hey @alan-baker please confirm you want me to allow any boolean answer for the NaN cases, even with this TODO.

Reply:
Changed to allow either answer.
Will file a post-v1 request for float-controls feature to allow preservation of infinities and nans at least.